docs/examples/language_model/language_model.ipynb

```diff
@@ -439,23 +459,263 @@
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Use pretrained state-of-the-art language models"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Use AWD LSTM language model\n",
+    "\n",
+    "AWD LSTM language model is the state-of-the-art RNN language model [1]. The main technique is to add weight-dropout on the recurrent hidden to hidden matrices to prevent the overfitting from occurring on the recurrent connections."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Load vocabulary and pretrained model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "AWDRNN(\n",
+      "  (embedding): HybridSequential(\n",
+      "    (0): Embedding(33278 -> 400, float32)\n",
+      "    (1): Dropout(p = 0.65, axes=(0,))\n",
+      "  )\n",
+      "  (encoder): Sequential(\n",
+      "    (0): LSTM(400 -> 1150.0, TNC)\n",
+      "    (1): LSTM(1150 -> 1150.0, TNC)\n",
+      "    (2): LSTM(1150 -> 400.0, TNC)\n",
+      "  )\n",
+      "  (decoder): HybridSequential(\n",
+      "    (0): Dense(400 -> 33278, linear)\n",
+      "  )\n",
+      ")\n",
+      "Vocab(size=33278, unk=\"<unk>\", reserved=\"['<eos>']\")\n"
+     ]
+    }
+   ],
+   "source": [
+    "awd_model_name = 'awd_lstm_lm_1150'\n",
+    "awd_model, vocab = nlp.model.get_model(awd_model_name, vocab=vocab, dataset_name=dataset_name, \n",
+    "                                   pretrained=True, ctx=context[0])\n",
+    "print(awd_model)\n",
+    "print(vocab)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Evaluate the pretrained model on val and test datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Best validation loss 4.30, val ppl 73.44\n",
+      "Best test loss 4.25, test ppl 69.81\n"
+     ]
+    }
+   ],
+   "source": [
+    "val_L = evaluate(awd_model, val_data, batch_size, context[0])\n",
+    "test_L = evaluate(awd_model, test_data, batch_size, context[0])\n",
+    "print('Best validation loss %.2f, val ppl %.2f'%(val_L, math.exp(val_L)))\n",
+    "print('Best test loss %.2f, test ppl %.2f'%(test_L, math.exp(test_L)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Use Cache LSTM language model\n",
+    "\n",
+    "Cache LSTM language model [2] adds a cache-like memory to neural network language models, e.g., AWD LSTM language model. It exploits the hidden outputs to define a probability distribution over the words in the cache. It generates the state-of-the-art results in inference time.\n",
+    "\n",
+    "![cache-model](cache_model.png)"
```

Comment:
specify width so that it doesn't take the whole screen.

Reply:
the bptt figure should also be with smaller width. fixed.